edge-hub/watchdog/linux/src/main.rs

```diff
@@ -0,0 +1,96 @@
+extern crate nix;
+
+use std::{error::Error, thread};
+use std::process::{Command, Stdio, Child};
+use std::sync::{Arc};
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::time::Duration;
+use nix::unistd::Pid;
+use nix::sys::signal::{self, Signal};
+use signal_hook::{iterator::Signals, SIGTERM};
+use tracing::{info};
+use tracing_subscriber;
+
+fn main() -> Result<(), Box<dyn Error>> {
```

Comment:
I think this Box<dyn Error> is necessitated by Signals, but need to double check.